### PR TITLE
docs: update supported platforms

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by clason** [comment]
> Tier 3: Not tested and no guarantees, but may work.

"and not all features may work". (Still don't think that qualifies as a "supported platform"; I'd just remove the line -- and the Tier, if you prefer -- but keep the note if you must.

**Review by justinmk — APPROVED** [review]

File: runtime/doc/support.txt [hunk 1]
```diff
@@ -13,12 +13,13 @@ Supported platforms					 *supported-platforms*
 
 `System`          `Tier`      `Versions`                  `Tested versions`
 Linux            1      >= 2.6.32, glibc >= 2.12     Ubuntu 22.04
-macOS (Intel)    1      >= 10.15                     macOS 12
-Windows 64-bit   1      >= 8 (see note below)        Windows Server 2022
+macOS (Intel)    1      >= 11                        macOS 12
+macOS (M1)       2      >= 11                        macOS 14
```
Comment (zeertzjq) [inline runtime/doc/support.txt:17]: Why do macOS (Intel) and macOS (M1) have different support tiers?

Comment (clason) [reply runtime/doc/support.txt:17]: Because we only test(ed) on Intel before M1 runners were GA.

Comment (dundargoc) [reply runtime/doc/support.txt:17]: Yeah, probably just outdated. Send PR

Comment (clason) [reply runtime/doc/support.txt:17]: (and chocolate.)

**Pull request merged**